content/docs/concepts/oakestra-extensions/addons.md

```diff
@@ -0,0 +1,87 @@
+- An **extension**, on the other hand, enhances the system capabilities by adding optional features without altering any core component.
+
+To simplify, Oakestra refers to both plugins and extensions collectively as **addons**, a term that represents any additional component that can enhance or modify Oakestras capabilities.
+
+### Plugin Example
+
+In Oakestra, plugins provide a way to swap out components, enhancing the systems adaptability. For example, the default scheduler could be replaced with an alternative that uses a more advanced algorithm. Consider a scheduler that accounts for environmental variables, such as energy consumption or green energy sources, when making scheduling decisions. This plugin can be enabled by users who need these advanced features, offering them a customized scheduling tool while other users retain the default, core scheduler. This approach keeps the core system minimal while catering to varied user requirements.
+
+### Extensions Example
+
+Extensions in Oakestra can add substantial new capabilities to the system. One example could be the introduction of federated learning on edge computing devices. Federated learning enables decentralized training across multiple devices, enhancing privacy by not requiring raw data transfer. Implementing this as an extension involves adding new components to the Orchestrator control plane for the coordination and aggregation of distributed training processes. Users can activate or deactivate this extension based on their needs, making it a flexible addition that adapts Oakestra to specialized use cases.
+
+## Addon System Design
+
+Addons in Oakestra are modular components that extend the platforms core functionality while keeping the foundational architecture lightweight. This design ensures that Oakestra remains efficient, avoiding unnecessary bloat in edge computing environments.
+
+Addons system is composed of two subsystems:
+- **Addons Marketplace**: Consists of Marketplace Manager and MongoDB client.
+- **Addons Engine**: Consists of Addons Manager, Addons Monitor, and MongoDB Client.
+
+#### Addons Marketplace:
+
+![](marketplace_light.png)
+
+The marketplace allows developers to publish their add-ons for Oakestra. To create
+an addon, users must package their code and convert it into an OCI-compliant image that is
+publicly accessible. Currently, only docker images are supported, but it should be possible to
+use any OCI-compliant image in the future. The Addons Marketplace subsystem offers API
+endpoints for developers to publish their addons to Oakestra. Upon receiving a request for
+registering an Addon, the marketplace asynchronously checks if the image is valid. Once
+it’s verified, the addon’s state in the database changes from `under_review` to `approved`. If
+invalid, the status becomes `failed_verification`.
+
+#### Addons Engine:
+
+![](engine_light.png)
```

Comment:
Images dont show up in light/dark mode. Please either see @Malyuk-A work on customer SVGs (https://github.com/nitindermohan/oak-new-docs/pull/13) or see built-in Doks functionality: https://getdoks.org/docs/basics/shortcodes/#assets-directory-1